aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,373 @@
+				Optional: true,
+			},
+			"health_check_protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  globalaccelerator.HealthCheckProtocolTcp,
+				ValidateFunc: validation.StringInSlice([]string{
+					globalaccelerator.HealthCheckProtocolTcp,
+					globalaccelerator.HealthCheckProtocolHttp,
+					globalaccelerator.HealthCheckProtocolHttps,
+				}, false),
+			},
+			"threshold_count": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      3,
+				ValidateFunc: validation.IntBetween(1, 10),
+			},
+			"traffic_dial_percentage": {
+				Type:         schema.TypeFloat,
+				Optional:     true,
+				Default:      100,
+				ValidateFunc: validation.IntBetween(0, 100),
+			},
+			"endpoint_configuration": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 10,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"endpoint_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"weight": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+	region := meta.(*AWSClient).region
+
+	opts := &globalaccelerator.CreateEndpointGroupInput{
+		ListenerArn:         aws.String(d.Get("listener_arn").(string)),
+		IdempotencyToken:    aws.String(resource.UniqueId()),
+		EndpointGroupRegion: aws.String(region),
+	}
+
+	if v, ok := d.GetOk("endpoint_group_region"); ok {
+		opts.EndpointGroupRegion = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("health_check_interval_seconds"); ok {
+		opts.HealthCheckIntervalSeconds = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_path"); ok {
+		opts.HealthCheckPath = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("health_check_port"); ok {
+		opts.HealthCheckPort = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("health_check_protocol"); ok {
+		opts.HealthCheckProtocol = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("threshold_count"); ok {
+		opts.ThresholdCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("traffic_dial_percentage"); ok {
+		opts.TrafficDialPercentage = aws.Float64(v.(float64))
+	}
+
+	if v, ok := d.GetOk("endpoint_configuration"); ok {
+		opts.EndpointConfigurations = resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Create Global Accelerator endpoint group: %s", opts)
+
+	resp, err := conn.CreateEndpointGroup(opts)
+	if err != nil {
+		return fmt.Errorf("Error creating Global Accelerator endpoint group: %s", err)
+	}
+
+	d.SetId(*resp.EndpointGroup.EndpointGroupArn)
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		err = resourceAwsGlobalAcceleratorAcceleratorWaitForState(conn, acceleratorArn)
```

Comment:
Sorry if my original comment was unclear. This waiter should exist outside this error conditional and replace the duplicated `StateChangeConf` logic below. e.g.

```go
acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())

if err != nil {
	return err
}

err = resourceAwsGlobalAcceleratorAcceleratorWaitForState(conn, acceleratorArn)

if err != nil {
	return err
}

return resourceAwsGlobalAcceleratorEndpointGroupRead(d, meta)
```